packages/sdk/src/oracle/store/index.ts

```diff
@@ -0,0 +1,18 @@
+import type { State } from "../types/state";
+import Write from "./write";
+import produce from "immer";
+
+export type WriteCallback = (write: Write) => void;
+export type Store = {
+  write: (cb: WriteCallback) => void;
+};
+
+export function factory(emit: (state: State, prev: State) => void = () => ({}), state: State = {}): Store {
+  return {
+    write(cb: WriteCallback) {
+      const prevState = state;
+      state = produce(state, (draft: State) => cb(new Write(draft)));
+      emit(state, prevState);
+    },
```

Comment:
Could you add some comments around what this is doing? These steps seem a little dense.

Reply:
added some comments around what this does